### PR TITLE
MODKBEKBJ-412 - Support DELETE /eholdings/kb-credentials/{credId}/users/{userId} to remove association between user and KB

### PR DESCRIPTION
**Pull request by natali-zaitseva**
## Purpose
implementation story for https://issues.folio.org/browse/MODKBEKBJ-412

## Approach
* modified ModuleDescriptor file
* modified assigned-users.raml file
* added implementation for DELETE /eholdings/kb-credentials/{credId}/users/{userId}
* added tests

### CODE REVIEW

File: src/main/java/org/folio/repository/assigneduser/AssignedUsersConstants.java [hunk 1]
```diff
@@ -30,11 +36,13 @@
     String[] updateColumns = new String[] {
       USER_NAME, FIRST_NAME, MIDDLE_NAME, LAST_NAME, PATRON_GROUP
     };
+    ASSIGNED_USER_COLUMN_LIST = String.format(join(",", nCopies(7, "%s")), (Object[]) allColumns);
```
Comment (psmagin) [inline src/main/java/org/folio/repository/assigneduser/AssignedUsersConstants.java:39]: Don't see where this const is used.

File: src/test/java/org/folio/rest/impl/integrationsuite/EholdingsAssignedUsersImplTest.java [hunk 2]
```diff
@@ -152,56 +156,40 @@ public void shouldReturn422OnPostWhenAssignedUserDoesNotHaveRequiredParameters()
...
+    String endpoint = String.format(ASSIGN_USER_PATH, credentialsId);
+    Errors errors = postWithStatus(endpoint, postBody, SC_UNPROCESSABLE_ENTITY).as(Errors.class);
 
     assertThat(errors.getErrors(), hasSize(5));
     assertThat(errors.getErrors(), everyItem(hasProperty("message", equalTo("may not be null"))));
   }
 
   @Test
-  public void shouldReturn204OnPutWhenAssignedUserIsValid() {
+  public void shouldReturn204OnDeleteUserAssignment(){
```
Comment (psmagin) [inline src/test/java/org/folio/rest/impl/integrationsuite/EholdingsAssignedUsersImplTest.java:167]: This tests for `put` are needed.

File: src/test/java/org/folio/util/AssignedUsersTestUtil.java [hunk 3]
```diff
@@ -64,14 +61,6 @@ public static String insertAssignedUser(String credentialsId, String username, S
     return future.join();
   }
 
-  public static String resourcePath(String... params) {
```
Comment (psmagin) [inline src/test/java/org/folio/util/AssignedUsersTestUtil.java:67]: Why do you think that this method is not useful?

File: src/main/java/org/folio/repository/assigneduser/AssignedUserRepositoryImpl.java [hunk 4]
```diff
@@ -122,6 +125,27 @@
...
+    Promise<UpdateResult> promise = Promise.promise();
+    pgClient(tenant).execute(query, createParams(Arrays.asList(credentialsId, userId)), promise);
+
+    Future<UpdateResult> resultFuture = promise.future()
+      .recover(excTranslator.translateOrPassBy());
+    return mapResult(resultFuture, updateResult -> checkUserDeleted(userId, updateResult));
+  }
+
+  @Nullable
+  private Void checkUserDeleted(String userId, UpdateResult updateResult) {
```
Comment (psmagin) [inline src/main/java/org/folio/repository/assigneduser/AssignedUserRepositoryImpl.java:142]: This method also can be used in `update` method

File: src/main/java/org/folio/repository/assigneduser/AssignedUsersConstants.java (hunk 1 above)
Comment (natali-zaitseva) [reply src/main/java/org/folio/repository/assigneduser/AssignedUsersConstants.java]: removed

File: src/test/java/org/folio/rest/impl/integrationsuite/EholdingsAssignedUsersImplTest.java (hunk 2 above)
Comment (natali-zaitseva) [reply src/test/java/org/folio/rest/impl/integrationsuite/EholdingsAssignedUsersImplTest.java:167]: probably during the rebase I accidentally removed them. Thank you for note, added them back

File: src/test/java/org/folio/util/AssignedUsersTestUtil.java (hunk 3 above)
Comment (natali-zaitseva) [reply src/test/java/org/folio/util/AssignedUsersTestUtil.java:67]: it is not clear from my perspective what is the resource path in the current test is used and the method name does not help to understand the context of  resourcePath

File: src/main/java/org/folio/repository/assigneduser/AssignedUserRepositoryImpl.java (hunk 4 above)
Comment (natali-zaitseva) [reply src/main/java/org/folio/repository/assigneduser/AssignedUserRepositoryImpl.java]: added to `update` method

**Pull request merged**